packages/x-components/src/x-modules/queries-preview/components/query-preview-list.vue

```diff
@@ -48,37 +58,54 @@
     public animation!: Vue | string;
 
     /**
-     * List of queries to preview.
+     * The list of queries preview to render.
+     *
+     * @public
      */
     @Prop({ required: true })
-    public queries!: string[];
+    public queriesPreviewInfo!: QueryPreviewInfo[];
 
     /**
      * Contains the status of the preview requests, indexed by query.
      */
     public queriesStatus: QueryPreviewStatusRecord = {};
 
+    /**
+     * The list of queries to preview.
+     *
+     * @returns The list of queries in the queriesPreviewInfo list.
+     * @internal
+     */
+    protected get queries(): string[] {
+      return this.queriesPreviewInfo.map(item => item.query);
+    }
+
     /**
      * Gets all the queries to render, that are those that don't have an `error` status.
      *
      * @returns A list of queries.
      * @internal
      */
-    protected get renderedQueries(): string[] {
-      return this.queries.filter(
-        query => this.queriesStatus[query] === 'success' || this.queriesStatus[query] === 'loading'
+    protected get renderedQueryPreviews(): QueryPreviewInfo[] {
+      return this.queriesPreviewInfo.filter(
+        ({ query }) =>
+          this.queriesStatus[query] === 'success' || this.queriesStatus[query] === 'loading'
       );
     }
 
     /**
      * Resets the status of all queries if they change.
      *
+     * @param newQueries - The new queries.
+     * @param oldQueries - The old queries.
      * @internal
      */
     @Watch('queries', { immediate: true })
-    protected resetStatusRecord(): void {
-      this.queriesStatus = {};
-      this.loadNext();
+    protected resetStatusRecord(newQueries: string[], oldQueries: string[]): void {
+      if (newQueries?.toString() !== oldQueries?.toString()) {
```

Comment:
Changing the prop and watching a computed property creates false positives when pairing this component with the semantic queries. This prevents that. (Thinking of creating a better function that compares arrays and adding it to the arrays utils in `x-utils`)